client/src/app.js

```diff
@@ -148,6 +147,37 @@
       console.warn('ERROR(' + err.code + '): ' + err.message);
     }
 
+
+    var sessionHeroOrder = $window.JSON.parse($window.sessionStorage.getItem('heroOrder'));
+    if( sessionHeroOrder !== null ) {
+      heroFactory.setOrder(sessionHeroOrder);
+      console.log('default hero order loaded');
+    }
+
+    var sessionReqOrder = $window.JSON.parse($window.sessionStorage.getItem('requesterOrder'));
+    if( sessionReqOrder !== null ) {
+      requesterFactory.setOrder(sessionReqOrder);
+      console.log('default req order loaded');
+    }
+
+    $rootScope.$watch(function() {
+      return heroFactory.getOrder();
+    }, function watchCallback (newVal, oldVal){
+      var stringObject = $window.JSON.stringify(newVal);
+      $window.sessionStorage.setItem('heroOrder',stringObject);
+      console.log(newVal);
+    }, true);
+
+    $rootScope.$watch(function() {
+      return requesterFactory.getOrder();
+    }, function watchCallback (newVal, oldVal){

```

Comment:
no space after `watchCallback` need space after `(newVal, oldVal)`

it should be `watchCallback(newVal, oldVal) {`